source/PrintData.cpp

```diff
@@ -0,0 +1,473 @@
+/* PrintData.cpp
+Copyright (c) 2022 by warp-core
+
+Main function for Endless Sky, a space exploration and combat RPG.
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "PrintData.h"
+
+#include "GameData.h"
+#include "Outfit.h"
+#include "Ship.h"
+
+#include <iostream>
+#include <map>
+#include <set>
+
+using namespace std;
+
+
+
+void PrintData::Print(char *argv[])
+{
+	const char *const *it = argv + 2;
+	if(!*it)
+		return;
+	string arg = *it;
+	if(arg == "-s" || arg == "--ships")
+		Ships(argv);
+	else if(arg == "-w" || arg == "--weapons")
+		Weapons(argv);
+	else if(arg == "-e" || arg == "--engines")
+		PrintEngineStats();
+	else if(arg == "-p" || arg == "--power")
+		PrintPowerStats();
+	else if(arg == "-o" || arg == "--outfits")
+		Outfits(argv);
+	else if(arg == "-h" || arg == "--help")
+		Help();
+	cout.flush();
+}
+
+
+
+void PrintData::Help()
+{
+	cerr << "--printdata: follow with arguments to print diffferent data." << endl;
+	cerr << "    -s, --ships: prints table of various base stats of all ships (not variants)." << endl;
+	cerr << "        -d, --deterrences: prints ship names and deterrence values, not variants." << endl;
+	cerr << "        -v, --variants: includes variants." << endl;
+	cerr << "        -s, --sales: prints a list of ships and the outfitters they appear in." << endl;
+	cerr << "        -o, --old: the older version of the PrintShipTable() function." << endl;
+	cerr << "    -w, --weapons: prints table of weapons and various stats." << endl;
+	cerr << "        -d, --deterrences: prints list of weapons and deterrence values." << endl;
+	cerr << "    -e, --engines: prints table of engines and various stats." << endl;
+	cerr << "    -p, --power: prints table of power outfits and various stats." << endl;
+	cerr << "    -o, --outfits: prints a list of outfits." << endl;
+	cerr << "        -s, --sales: prints a list of outfits and all the outfitters they appear in." << endl;
```

Comment:
You could make `--sales` not be a nested argument so that if just `--sales` is provided shipyards and outfitters are printed. And if `--outfits` (or `--ships`) is provided then only that sale.

Although I'm not sure whether outputting both is even something that is useful. Might be better to simply provide `--outfitters` and `--shipyards`.

Reply:
I'd expect `--outfitters` or `--shipyards` to just provide a list of outfitters or shipyards and their contents (which could be added if desired). The 'sales' mode for outfits and ships here provide tables of outfits (or ships) where each item is followed by a list of outfitters or shipyards that it appears in. The table in #7105 shows this format (the 'designer/manufacturer' column was added manually by me in the spreadsheet).
Making it non-nested would mean it can't be aliased with `-s`, because that could also mean shpis.